tests/General/MainTest.php

```diff
@@ -4,7 +4,10 @@ class MainTest extends \PHPUnit\Framework\TestCase {
 
     public function testRouteResolving() {
         $router = new Route();
-        $this->assertEquals('http://tum.sexy/hunger', $router->getTargetOfSub('hunger'));
+        // Normal redirect
+        $this->assertEquals('http://tum.sexy/hunger', $router->getTargetOfSub('hunger', null));
+        // Subsubdomain redirect to moodle
+        $this->assertContains('moodle.tum.de', $router->getTargetOfSub('anal', 'm'));
```

Comment:
shouldn't we also verify the whole URL is generated correctly? Why only contains?

Reply:
Wanted it to keep flexible. For m.tum.sexy we redirect to the tum login for example and I thought about doing it for the new noodle redirects as well. But right, should probably test the complete URL.

Reply:
You can do both. Separate test cases.